googleapis_auth/lib/src/metadata_server_client.dart

```diff
@@ -39,8 +39,8 @@ Future<AccessCredentials> obtainAccessCredentialsViaMetadataServer(
 /// {@macro googleapis_auth_not_close_the_baseClient}
 Future<AutoRefreshingAuthClient> clientViaMetadataServer({
   Client? baseClient,
```

Comment:
Weird, I know – but you actually WANT this.

Otherwise you lose the API call in the async stack trace

Reply:
@kevmoo the analyser `unnecessary_await_in_return` picks this one up as an issue

Reply:
https://dart-lang.github.io/linter/lints/unnecessary_await_in_return.html

Mostly style.